src/com/facebook/buck/apple/project_generator/ProjectGenerator.java

```diff
@@ -1443,18 +1446,23 @@ private PBXNativeTarget generateBinaryTarget(
       }
     }
 
+    Optional<String> moduleName =
+        isModularAppleLibrary ? Optional.of(getProductName(targetNode)) : Optional.empty();
```

Comment:
`SwiftCompile` actually gets `args.getModuleName().orElse(buildTarget.getShortName())` as the module name, I think we should be consistent here. It might be best to centralise this logic in a static method on `SwiftLibraryDescription`.

Reply:
I'll consolidate this a little bit better in project gen so it is the same. As for consolidating between project gen and buck build I think we should change this by having a `modular` interface, so this logic can be shared between CxxLibrary and SwiftLibrary. This will allow us to implement it into `SwiftDescriptions.populateSwiftLibraryDescriptionArg` properly.